### DISCUSSION TITLE
Authorization Error on validation-report

### DISCUSSION THREAD

**Discussion by rengit62** [OP-TED/eForms-SDK#377] [Q&A] [answered]
Hi Labs !
I submitted some notices successfully by java but I get error authorization error on validation-report
1) code for submit :
*********************************************************
Request request = new Request.Builder()
				.url("https://enotices2.preview.ted.europa.eu/esenders/api/v2/notice/notices/submit")
				.header("accept", "application/json")
				.header("Authorization", BEARER_AUTH)
				.post(requestBody)
				.build();
		Response response = client.newCall(request).execute();
2) code for validationReport
final HttpHeaders headers = new HttpHeaders();
		headers.set("Authorization", BEARER_AUTH);
		headers.set("accept", "*/*");
		RestTemplate restTemplate = new RestTemplate();
		HttpEntity<String> httpEntity = new HttpEntity<>(headers);
		ResponseEntity<String> resp = null;
		try {
			resp = restTemplate.exchange(**url**,HttpMethod.GET,httpEntity,String.class);
		} catch (Exception e) {
			logger.error(e);
			throw e ;
		}
		return resp ;
error on validationReport : 
<?xml version="1.0" encoding="UTF-8"?><EOL><Error><Code>InvalidArgument</Code><Message>Unsupported Authorization Type</Message><ArgumentName>Authorization</ArgumentName><ArgumentValue>Bearer *******************************</ArgumentValue><RequestId>0FGMVXTRK651F7BS</RequestId><HostId>slWeKl38cQTU/ffEYCQuh/ucZHkdrxFdf14ikjdXCkZDILLcGyWS+lLW0l3c3Lus1G1y3UH3J9k=</HostId></Error>"

the **url** is in the response of https://enotices2.preview.ted.europa.eu/esenders/api/v2/notice/notices/submit"
{"validationReportUrl":"https://enotices2.preview.ted.europa.eu/esenders/notice/notices/11032cc1-c586-4f41-92bb-a74fed77920e-01/validation-report","success":false}

Can you help me ?
Thanks and Regards
Renato Baviera

**Comment by pcolast** [comment, marked as answer]
Hello Renato,

Response body for submit notice contains property validationReportUrl indicating a url for getting validation report. 
Currently there is a bug.
 
The validationReportUrl being returned by submit notice looks like
'https://enotices2.preview.ted.europa.eu/esenders/notice/notices/3fa85f64-5717-4562-b3fc-2c963f66afa6-01/validation-report'
whereas the correct value should be 
'https://enotices2.preview.ted.europa.eu/esenders/api/v2/notice/notices/3fa85f64-5717-4562-b3fc-2c963f66afa6-01/validation-report'
the **/api/v2/** is missing in the validationReportUrl 

Until this bug is fixed, we recommend rather use url pattern 'https://enotices2.preview.ted.europa.eu/esenders/api/v2/notice/notices/{businessId}/validation-report'
where the {businessId} is replaced with value taken from the submitted notice ('noticeId-versionId') or with {businessId} from validationReportUrl.

We apologize for the inconvenience.

Have a nice day
Best regards,
Stefan
